metadata/src/test/java/org/apache/kafka/controller/BootstrapMetadataTest.java

```diff
@@ -31,9 +35,21 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class BootstrapMetadataTest {
+    private Path tmpDir;
```

Comment:
Not sure we gain much by using `Path` since all our utility methods work with `File`.

Reply:
I think we should continue to use Path because:
1. `BootstrapMetadata.write()` used in every test in this file accepts a `Path`. 
2. Prior to this change, we were using Path in this test file, so the current PR doesn't change anything on that front.
3. The new Java NIO.2 introduced many new(aka better) helper methods using `Files.*` which use `Path` instead of files. I am already starting to replace the older `java.io` methods with new Nio.2 methods in the code base [1]. Due to the benefits of the new helper methods, I think we should continue to stick use `Path` instead of `File`.

However, please let me know if you feel strongly about this comment, I will make the change accordingly.

[1] https://issues.apache.org/jira/browse/KAFKA-13928